### PR TITLE
adds cones for felinids

### PR DESCRIPTION
**Pull request by Plyushsune**
meow meow

## About the PR 

**Media**

- [ ] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl:
- add: cones to wear
- remove: Removed fun!

### CODE REVIEW

**Comment by Elijahrane** [comment]
screenshot please

File: Resources/Prototypes/Catalog/Research/technologies.yml [hunk 1]
```diff
@@ -104,6 +104,7 @@
   - MedkitAdvanced
   - MedkitRadiation
   - MedkitCombat
+  - felinidcone
```
Comment (Elijahrane) [inline Resources/Prototypes/Catalog/Research/technologies.yml]: Casing

**Comment by Plyushsune** [comment]
> screenshot please

working on it, i just been busy

File: Resources/Prototypes/Catalog/VendingMachines/Inventories/medidrobe.yml [hunk 2]
```diff
@@ -18,3 +18,4 @@
     ClothingHeadHatParamedicsoft: 2
     ClothingHeadHelmetVoidParamed: 1
     ClothingOuterHardsuitVoidParamed: 1
+    felinidcone: 10
```
Comment (Elijahrane) [inline Resources/Prototypes/Catalog/VendingMachines/Inventories/medidrobe.yml]: Change casing (of the ID in masks.yml), also 10 is way too many. Should be 1

File: Resources/Prototypes/Nyanotrasen/Entities/Clothing/Masks/masks.yml [hunk 3]
```diff
@@ -28,3 +28,16 @@
     sprite: Nyanotrasen/Clothing/Mask/mask_oni_red.rsi
   - type: Clothing
     sprite: Nyanotrasen/Clothing/Mask/mask_oni_red.rsi
+
+- type: entity
+  parent: ClothingMaskPullableBase
+  id: felinidcone
+  name: Recovery cone
```
Comment (Elijahrane) [inline Resources/Prototypes/Nyanotrasen/Entities/Clothing/Masks/masks.yml]: casing for name would be "recovery cone". The game will capitalize it if it needs capitalized. Only capitalize proper nouns in the name field.

File: Resources/Prototypes/Nyanotrasen/Entities/Clothing/Masks/masks.yml [hunk 4]
```diff
@@ -28,3 +28,16 @@
     sprite: Nyanotrasen/Clothing/Mask/mask_oni_red.rsi
   - type: Clothing
     sprite: Nyanotrasen/Clothing/Mask/mask_oni_red.rsi
+
+- type: entity
+  parent: ClothingMaskPullableBase
+  id: felinidcone
```
Comment (Elijahrane) [inline Resources/Prototypes/Nyanotrasen/Entities/Clothing/Masks/masks.yml]: Casing for ID would be FelinidCone. The ID should also probably just be ClothingNeckFelinidCone and go in the neck slot, no?

File: Resources/Prototypes/Nyanotrasen/Entities/Clothing/Masks/masks.yml [hunk 5]
```diff
@@ -28,3 +28,16 @@
     sprite: Nyanotrasen/Clothing/Mask/mask_oni_red.rsi
   - type: Clothing
     sprite: Nyanotrasen/Clothing/Mask/mask_oni_red.rsi
+
+- type: entity
+  parent: ClothingMaskPullableBase
```
Comment (Elijahrane) [inline Resources/Prototypes/Nyanotrasen/Entities/Clothing/Masks/masks.yml:33]: Parent ClothingNeckBase. Unless this is really supposed to be a mask?

File: Resources/Prototypes/Nyanotrasen/Entities/Clothing/psionic_clothing.yml [hunk 6]
```diff
@@ -64,3 +64,19 @@
   - type: ClothingGrantPsionicPower
     power: NoosphericZapPower
   - type: Psionic
+
+- type: entity
+  parent: ClothingMaskPullableBase
```
Comment (Elijahrane) [inline Resources/Prototypes/Nyanotrasen/Entities/Clothing/psionic_clothing.yml]: This should parent ClothingNeckFelinidCone

File: Resources/Prototypes/Nyanotrasen/Entities/Clothing/Masks/masks.yml [hunk 7]
```diff
@@ -28,3 +28,16 @@
...
+  id: felinidcone
+  name: Recovery cone
+  description: A recovery cone adapted for felinids and other animal-based morthotypes.
+  abstract: true
```
Comment (Elijahrane) [inline Resources/Prototypes/Nyanotrasen/Entities/Clothing/Masks/masks.yml]: Not abstract

File: Resources/Prototypes/Nyanotrasen/Entities/Clothing/Masks/masks.yml [hunk 8]
```diff
@@ -28,3 +28,16 @@
     sprite: Nyanotrasen/Clothing/Mask/mask_oni_red.rsi
   - type: Clothing
     sprite: Nyanotrasen/Clothing/Mask/mask_oni_red.rsi
+
+- type: entity
+  parent: ClothingMaskPullableBase
+  id: felinidcone
+  name: Recovery cone
+  description: A recovery cone adapted for felinids and other animal-based morthotypes.
```
Comment (Elijahrane) [inline Resources/Prototypes/Nyanotrasen/Entities/Clothing/Masks/masks.yml]: They're all animals

File: Resources/Prototypes/Nyanotrasen/Entities/Clothing/Masks/masks.yml [hunk 9]
```diff
@@ -28,3 +28,16 @@
...
+  abstract: true
+  components:
```
Comment (Elijahrane) [inline Resources/Prototypes/Nyanotrasen/Entities/Clothing/Masks/masks.yml:37]: Can probably use some disease resistance (like 0.05)

**Review by Elijahrane — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/Nyanotrasen/Entities/Clothing/Masks/masks.yml (hunk 5 above)
Comment (Elijahrane) [reply Resources/Prototypes/Nyanotrasen/Entities/Clothing/Masks/masks.yml:33]: Needless to say if you do this this also needs moved to Clothing/Neck/

File: Resources/Textures/Nyanotrasen/Clothing/Mask/felinidcone.rsi/meta.json [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+    "states": [
+      {
+        "name": "icon"
+      },
+      {
+        "name": "inhand-left",
+        "directions": 4
+      },
+      {
+        "name": "inhand-right",
```
Comment (Elijahrane) [inline Resources/Textures/Nyanotrasen/Clothing/Mask/felinidcone.rsi/meta.json:18]: just drop the inhand states if there's no sprite for it. delete the 2 inhand pngs and delete the states here

File: Resources/Prototypes/Catalog/Research/technologies.yml (hunk 1 above)
Comment (Elijahrane) [reply Resources/Prototypes/Catalog/Research/technologies.yml]: You also need to make a lathe prototype with its cost, etc for this to work. This isn't looking for the entity proto, it's looking for the lathe recipe proto so it knows how many materials to charge, etc

File: Resources/Textures/Nyanotrasen/Clothing/Mask/felinidcone.rsi/meta.json (hunk 10 above)
Comment (Plyushsune) [reply Resources/Textures/Nyanotrasen/Clothing/Mask/felinidcone.rsi/meta.json:18]: there are inhand sprites

Comment (Elijahrane) [reply Resources/Textures/Nyanotrasen/Clothing/Mask/felinidcone.rsi/meta.json:18]: my bad

File: Resources/Prototypes/Nyanotrasen/Entities/Clothing/Masks/masks.yml (hunk 4 above)
Comment (Plyushsune) [reply Resources/Prototypes/Nyanotrasen/Entities/Clothing/Masks/masks.yml]: i intend it to block eating and drinking

Comment (Elijahrane) [reply Resources/Prototypes/Nyanotrasen/Entities/Clothing/Masks/masks.yml]: does ingestionblocker care about which slot it's in...?

Comment (Plyushsune) [reply Resources/Prototypes/Nyanotrasen/Entities/Clothing/Masks/masks.yml]: *shrugs*

Comment (Plyushsune) [reply Resources/Prototypes/Nyanotrasen/Entities/Clothing/Masks/masks.yml]: i fixed casing but it doesn't want me to close this oddly

**Comment by Plyushsune** [comment]
image
image
image
image

**Comment by Plyushsune** [comment]
image
updated sprites

File: Resources/Prototypes/Nyanotrasen/Entities/Clothing/Masks/masks.yml [hunk 11]
```diff
@@ -28,3 +28,18 @@
     sprite: Nyanotrasen/Clothing/Mask/mask_oni_red.rsi
   - type: Clothing
     sprite: Nyanotrasen/Clothing/Mask/mask_oni_red.rsi
+
+- type: entity
+  parent: ClothingMaskPullableBase
+  id: FelinidCone
```
Comment (Vordenburg) [inline Resources/Prototypes/Nyanotrasen/Entities/Clothing/Masks/masks.yml]: If this descends from a clothing prototype, I'd prefer it follows the convention of other children. I.e. there was a commit pulled in the recent cherrypick that made `HatBandBlack` etc. into `ClothingHeadBandBlack`.

In this case, you'd make it `ClothingMaskFelinidCone`.

That said, this probably makes sense in a neck slot, but if it's in a mask slot, it'd block wearing gas masks too which seems to make sense given the bulk of the item?

I'm not sure on this one. If it's a mask, it shouldn't be pullable, at least. Those things probably aren't easy to remove.

File: Resources/Prototypes/Nyanotrasen/Entities/Clothing/Masks/masks.yml [hunk 12]
```diff
@@ -28,3 +28,18 @@
...
+  name: recovery cone
+  description: A recovery cone adapted for felinids.
+  abstract: false
```
Comment (Vordenburg) [inline Resources/Prototypes/Nyanotrasen/Entities/Clothing/Masks/masks.yml]: Abstract is false by default; you can remove this line.

File: Resources/Prototypes/Nyanotrasen/Entities/Clothing/psionic_clothing.yml [hunk 13]
```diff
@@ -64,3 +64,19 @@
   - type: ClothingGrantPsionicPower
     power: NoosphericZapPower
   - type: Psionic
+
+- type: entity
+  parent: FelinidCone
+  id: felinidConePsionic
+  name: ZE CONE OF ALL-SEEING
+  description: CONE CONE!!!?!??!?!??!?!!
+  abstract: false
```
Comment (Vordenburg) [inline Resources/Prototypes/Nyanotrasen/Entities/Clothing/psionic_clothing.yml]: See comment about `abstract: false`.

File: Resources/Prototypes/Nyanotrasen/Entities/Clothing/psionic_clothing.yml [hunk 14]
```diff
@@ -64,3 +64,19 @@
   - type: ClothingGrantPsionicPower
     power: NoosphericZapPower
   - type: Psionic
+
+- type: entity
+  parent: FelinidCone
+  id: felinidConePsionic
```
Comment (Vordenburg) [inline Resources/Prototypes/Nyanotrasen/Entities/Clothing/psionic_clothing.yml]: Casing: `FelinidConePsionic`

File: Resources/Prototypes/Nyanotrasen/Entities/Clothing/psionic_clothing.yml [hunk 15]
```diff
@@ -64,3 +64,19 @@
...
+  name: ZE CONE OF ALL-SEEING
+  description: CONE CONE!!!?!??!?!??!?!!
```
Comment (Vordenburg) [inline Resources/Prototypes/Nyanotrasen/Entities/Clothing/psionic_clothing.yml]: This name and description is too silly.

File: Resources/Prototypes/Nyanotrasen/Entities/Clothing/psionic_clothing.yml [hunk 16]
```diff
@@ -64,3 +64,19 @@
...
+  abstract: false
+  components:
+  - type: IngestionBlocker
+  - type: ClothingGrantPsionicPower
+    power: TelegnosisPower
+  - type: Psionic
```
Comment (Vordenburg) [inline Resources/Prototypes/Nyanotrasen/Entities/Clothing/psionic_clothing.yml:77]: If you're going to make psionic clothing, you should probably put it in the PsionicArtifactPool.

File: Resources/Prototypes/Recipes/Lathes/medical.yml [hunk 17]
```diff
@@ -173,3 +173,11 @@
   completetime: 2
   materials:
     Plastic: 300
+
+- type: latheRecipe
+  id: FelinidCone
+  result: FelinidCone
+  completetime: 2
+  materials:
+    Steel: 50
```
Comment (Vordenburg) [inline Resources/Prototypes/Recipes/Lathes/medical.yml:199]: I was going to say you could use plastic for this, until I read [this](https://en.wikipedia.org/wiki/Elizabethan_collar#Dangers_of_Elizabethan_collars).

File: Resources/Prototypes/Catalog/Research/technologies.yml [hunk 18]
```diff
@@ -104,6 +104,7 @@
   - MedkitAdvanced
   - MedkitRadiation
   - MedkitCombat
+  - FelinidCone
```
Comment (Vordenburg) [inline Resources/Prototypes/Catalog/Research/technologies.yml]: The R&D rework obsoleted this file, so you can just move it to a static recipe on the medical techfab.

**Review by Vordenburg — CHANGES_REQUESTED** [review]

**Comment by Vordenburg** [comment]
This PR makes me consider if a component that restricts usage by morphotype might be useful.
I'm also thinking now maybe switching it to a neck slot item might fix the back sprite layering issue. It's worth a try.

File: Resources/Prototypes/Nyanotrasen/Entities/Clothing/psionic_clothing.yml (hunk 15 above)
Comment (Plyushsune) [reply Resources/Prototypes/Nyanotrasen/Entities/Clothing/psionic_clothing.yml]: That's the point

File: Resources/Prototypes/Catalog/Research/technologies.yml (hunk 18 above)
Comment (Plyushsune) [reply Resources/Prototypes/Catalog/Research/technologies.yml]: The techfab locations don't seem to be straight forward, I might just fallback to just adding it to medidrobe

**Comment by Plyushsune** [comment]
That might be helpful for this

File: Resources/Prototypes/Catalog/Research/technologies.yml (hunk 18 above)
Comment (Vordenburg) [reply Resources/Prototypes/Catalog/Research/technologies.yml]: What is not straightforward about it? All of the lathes are in the two `Structures/Machines/lathe.yml` files. There's no reason why you couldn't put it there.

https://github.com/Nyanotrasen/Nyanotrasen/blob/b026d8d17437e72d54866df67cd717fe0583cfc9/Resources/Prototypes/Entities/Structures/Machines/lathe.yml#L371-L394

Comment (Plyushsune) [reply Resources/Prototypes/Catalog/Research/technologies.yml]: It's not in there?

Comment (Vordenburg) [reply Resources/Prototypes/Catalog/Research/technologies.yml]: Merge master.

**Comment by Plyushsune** [comment]
I'm not home at the moment.

File: Resources/Prototypes/Recipes/Lathes/medical.yml (hunk 17 above)
Comment (Plyushsune) [reply Resources/Prototypes/Recipes/Lathes/medical.yml:199]: in this case, it is a plastic cone

**Comment by Plyushsune** [comment]
all the errors created are stuff, i don't know how to fix

File: Resources/Prototypes/Recipes/Lathes/medical.yml [hunk 19]
```diff
@@ -191,9 +191,19 @@
     Plastic: 300
 
 - type: latheRecipe
+<<<<<<< HEAD
```
Comment (Vordenburg) [inline Resources/Prototypes/Recipes/Lathes/medical.yml:194]: This is a merge conflict marker left behind in the file. This is unfortunately not the way to resolve a merge conflict. It seems there's a lot going on in this PR now that doesn't quite make sense. Why is HolosignSystem being deleted, for example?

**Comment by Plyushsune** [comment]
image
how i feel closing the pr even though i can save it but im just too lazy to...

**Pull request closed without merging**